metadata.go

```diff
@@ -0,0 +1,90 @@
+package druid
+
+import (
+	"context"
+	_ "embed"
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/h2oai/go-druid/builder/query"
+)
+
+type count struct {
+	Cnt int `json:"cnt"`
+}
+
+// MetadataService is a service that runs druid metadata requests using druid SQL API.
+type MetadataService struct {
+	client             *Client
+	tickerMilliseconds time.Duration
+	waitTimeoutSeconds int
+}
+
+//go:embed sql/datasource_available.sql
+var datasourceAvailableQuery string
+
+func fillDataSourceName(in string, ds string) string {
+	return strings.Replace(in, "${{ datasource }}", ds, 1)
+}
+
+// AwaitDataSourceAvailable awaits for a datasource to be visible in druid table listing.
+func (md *MetadataService) AwaitDataSourceAvailable(dataSourceName string) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() {
+		time.Sleep(time.Duration(md.waitTimeoutSeconds) * time.Second)
+		cancel()
+	}()
+	ticker := time.NewTicker(md.tickerMilliseconds * time.Millisecond)
```

Comment:
We should also stop the ticker. I suggest defer.

Reply:
Done! Thanks.